src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -57,25 +57,36 @@ TestContext sContext;
 
 const char PAYLOAD[] = "Hello!";
 
-int gSendMessageCount = 0;
-
 class OutgoingTransport : public Transport::Base
 {
 public:
-    /// Transports are required to have a constructor that takes exactly one argument
-    CHIP_ERROR Init(const char * unused) { return CHIP_NO_ERROR; }
-
     CHIP_ERROR SendMessage(const PeerAddress & address, System::PacketBufferHandle && msgBuf) override
     {
-        gSendMessageCount++;
+        mSendMessageCount++;
+
+        if (!mDropNextMessage)
+        {
+            System::PacketBufferHandle receivedMessage = msgBuf.CloneData();
+            HandleMessageReceived(address, std::move(receivedMessage));
+        }
+        else
+        {
+            mDropNextMessage = false;
+            mDroppedMessageCount++;
+        }
 
         return CHIP_NO_ERROR;
     }
 
     bool CanSendToPeer(const PeerAddress & address) override { return true; }
+
+    bool mDropNextMessage         = false;
```

Comment:
nit: would it be useful to have a 'drop count' instead of a bool to validate that we can survive more than one dropped packet? This means one would not need to keep resetting mDropNextMessage.